deepsensor/data/loader.py

```diff
@@ -797,6 +829,66 @@ def sample_offgrid_aux(
             Y_t_aux = Y_t_aux.reshape(1, *Y_t_aux.shape)
         return Y_t_aux
 
+    def sample_patch_size_extent(self) -> Sequence[float]:
+        """Sample patch size.
+
+        :return sequence of patch spatial extent as [lat_min, lat_max, lon_min, lon_max]
+        """
+        # assumption of normalized spatial coordinates between 0 and 1
```

Comment:
We can't assume data is bounded in [0, 1]. This is not guaranteed or enforced in any part of the DeepSensor data processing pipeline. Instead, we need a new method, run during the TaskLoader `init`, which computes the global min/max coordinate values of the context/target data, and then the central point of the patch should be sampled uniformly in this range.

Reply:
Okay, to my understanding the TaskLoader only works on already normalized/standardized data and the coordinate bounds were normalized to [0,1] but that is good to know, thanks!

Reply:
By default, the `DataProcessor` linearly normalises the coords of the first data variable it is provided with to lie in `[0, 1]`, but subsequent variables may exceed that data range. Thus, although the data coords will typically lie in `[0, 1]`, there is nothing constraining this to always hold.